spec/components/candidate_interface/review_rejection_reasons_component_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'rails_helper'
+
+RSpec.describe CandidateInterface::ReviewRejectionReasonsComponent do
+  it 'renders a single rejection reason with a label' do
+    rejection_details = { id: 'quality_of_writing_details', text: 'Quality Bad' }
+    rejection_reasons = [RejectionReasons::Reason.new(id: 'quality_of_writing', label: 'Quality of writing', details: rejection_details)]
+
+    result = render_inline(described_class.new(rejection_reasons))
+
+    expect(result.css('p').text).to include('Quality of writing:Quality Bad')
+  end
+
+  it 'renders multiple rejections reason with a label' do
+    qualitfy_rejection_details = { id: 'quality_of_writing_details', text: 'Quality Bad' }
+    personal_statement_rejection_details = { id: 'personal_statement_other_details', text: 'Personal statement Bad' }
+    rejection_reasons = [
+      RejectionReasons::Reason.new(id: 'quality_of_writing', label: 'Quality of writing', details: qualitfy_rejection_details),
+      RejectionReasons::Reason.new(id: 'personal_statement_other', label: 'Other', details: personal_statement_rejection_details),
+    ]
+
+    result = render_inline(described_class.new(rejection_reasons))
+
+    expect(result.css('p').text).to include('Quality of writing:Quality Bad')
+    expect(result.css('p').text).to include('Other:Personal statement Bad')
+  end
+
+  it 'does not render the other label for a single rejection reason' do
+    personal_statement_rejection_details = { id: 'personal_statement_other_details', text: 'Personal statement Bad' }
+    rejection_reasons = [
+      RejectionReasons::Reason.new(id: 'personal_statement_other', label: 'Other', details: personal_statement_rejection_details),
+    ]
+
+    result = render_inline(described_class.new(rejection_reasons))
+
+    expect(result.css('p').text).not_to include('Other:')
+    expect(result.css('p').text).to include('Personal statement Bad')
+  end
+
+  describe '#hide_other_label?' do
+    it 'returns true if there is a single other rejection reasons' do
+      personal_statement_rejection_details = { id: 'personal_statement_other_details', text: 'Personal statement Bad' }
+      rejection_reasons = [
+        RejectionReasons::Reason.new(id: 'personal_statement_other', label: 'Other', details: personal_statement_rejection_details),
+      ]
+
+      result = described_class.new(rejection_reasons)
+
+      expect(result.hide_other_label?).to be(true)
```

Comment:
`expect(result).to hide_other_label`

Reply:
Attempting to use predicate matchers yields a `NameError: undefined local variable or method `hide_other_label'` error due to it not being prepended by 'be_` or `have_` (see the [matcher regex](https://github.com/rspec/rspec-expectations/blob/v3.11.0/lib/rspec/matchers.rb#L960))which doesn't read well with this method. I think using `be(true)` here is not risky as this method returns a boolean. Let me know your thoughts

Reply:
oh right, sorry, i should've checked it locally before requesting. happy with your suggestion.